packages/flutter/lib/src/rendering/table_border.dart

```diff
@@ -97,34 +97,147 @@ class TableBorder {
+        selector(bottom) == topValue &&
+        selector(left) == topValue;
+  }
+
+  /// Gets the set of distinct visible colors from the exterior border sides.
+  Set<Color> _getDistinctVisibleOuterColors() {
+    return <Color>{
+      if (top.style != BorderStyle.none) top.color,
+      if (right.style != BorderStyle.none) right.color,
+      if (bottom.style != BorderStyle.none) bottom.color,
+      if (left.style != BorderStyle.none) left.color,
+    };
+  }
+
+  void _paintTableBorder(Canvas canvas, Rect rect) {
+    if (outerBorderIsUniform && borderRadius != BorderRadius.zero) {
+      final RRect outer = borderRadius.toRRect(rect);
+      final RRect inner = outer.deflate(top.width);
+      final Paint paint = Paint()..color = top.color;
+      canvas.drawDRRect(outer, inner, paint);
+      return;
     }
 
-    final BorderStyle topStyle = top.style;
-    if (right.style != topStyle ||
-        bottom.style != topStyle ||
-        left.style != topStyle ||
-        horizontalInside.style != topStyle ||
-        verticalInside.style != topStyle) {
-      return false;
+    final Set<Color> visibleColors = _getDistinctVisibleOuterColors();
+    if (visibleColors.length == 1 && borderRadius != BorderRadius.zero) {
+      _paintNonUniformBorderWithRadius(
+        canvas,
+        rect,
+        borderRadius: borderRadius,
+        top: top.style == BorderStyle.none ? BorderSide.none : top,
+        right: right.style == BorderStyle.none ? BorderSide.none : right,
+        bottom: bottom.style == BorderStyle.none ? BorderSide.none : bottom,
+        left: left.style == BorderStyle.none ? BorderSide.none : left,
+        color: visibleColors.first,
+      );
+      return;
     }
 
-    return true;
+    paintBorder(canvas, rect, top: top, right: right, bottom: bottom, left: left);
+  }
+
+  /// Paints a non-uniform border with border radius support.
+  ///
+  /// This is similar to [BoxBorder.paintNonUniformBorder] but adapted for table borders.
+  void _paintNonUniformBorderWithRadius(
+    Canvas canvas,
+    Rect rect, {
+    required BorderRadius borderRadius,
+    BorderSide top = BorderSide.none,
+    BorderSide right = BorderSide.none,
+    BorderSide bottom = BorderSide.none,
+    BorderSide left = BorderSide.none,
+    required Color color,
+  }) {
+    final RRect borderRect = borderRadius.toRRect(rect);
+    final Paint paint = Paint()..color = color;
+
+    final RRect inner = _deflateRRect(
+      borderRect,
+      EdgeInsets.fromLTRB(left.strokeInset, top.strokeInset, right.strokeInset, bottom.strokeInset),
+    );
+    final RRect outer = _inflateRRect(
+      borderRect,
+      EdgeInsets.fromLTRB(
+        left.strokeOutset,
+        top.strokeOutset,
+        right.strokeOutset,
+        bottom.strokeOutset,
+      ),
+    );
+    canvas.drawDRRect(outer, inner, paint);
+  }
+
+  /// Inflates an [RRect] by the given [EdgeInsets].
+  static RRect _inflateRRect(RRect rect, EdgeInsets insets) {
```

Comment:
Are _inflateRect and _deflateRect here the same as the methods defined in BoxBorder? Perhaps those should be made reusable and `@protected` in the base class.

Reply:
I've created a new `RRectUtils` utility class to address the code duplication between `TableBorder` and `BoxBorder`. Since these classes have different base classes (`TableBorder` is standalone while `BoxBorder` extends `ShapeBorder`), we can't easily share the methods through inheritance.

The new utility class provides static methods for inflating and deflating RRects, which are now used by both `TableBorder` and `BoxBorder`. This eliminates code duplication while maintaining the existing class hierarchy.

I've also added comprehensive tests for the `RRectUtils` class to ensure it works correctly in all scenarios.